### PR TITLE
Add shared configurations to Root ESLint Plugin

### PR DESCRIPTION
**Pull request by williams4027**
Creates centrally shared set of eslint configurations based on combining the `.eslintrc.js` files of `root-web`, `root-mobile`, and `root-server`.

Followup PRs to change configurations (since `package.json` depends on a git commit hash for pulling the plugin):
https://github.com/Root-App/root-web/pull/267
https://github.com/Root-App/root-mobile/pull/3491
https://github.com/Root-App/root-server/pull/6776

### CODE REVIEW

File: lib/configs/recommended.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    'react/jsx-indent-props': ['error', 2],
+    'react/jsx-indent': ['error', 2],
+    'react/jsx-max-props-per-line': ['error', { 'maximum': 1 }],
+    'react/no-deprecated': ['error'],
+    'react/jsx-tag-spacing': ['error', { 'closingSlash': 'never', 'beforeSelfClosing': 'always', 'afterOpening': 'never' }],
+    'react/jsx-wrap-multilines': ['error'],
+    'react/self-closing-comp': 'error',
+    'react/sort-prop-types': ['error', { 'ignoreCase': true }],
+    'rest-spread-spacing': 'error',
+    'root/await-poll-for-condition': 'error',
```
Comment (TAGraves) [inline lib/configs/recommended.js]: This rule can be replaced by adding `pollForCondition` to the `root/preceded-by-await` rule.

**Review by TAGraves — APPROVED** [review]
This is great! Do you mind adding CI to this project as a follow-up?

File: lib/configs/recommended.js (hunk 1 above)
Comment (williams4027) [reply lib/configs/recommended.js]: Updated and removed the await-poll-for-condition rule as it is obsolete (added tests as well).

**Comment by williams4027** [comment]
Yes, I can add some CI to run the tests as a followup in [the trello card](https://trello.com/c/pXbJA8d0/49-pull-our-shared-eslint-config-out-into-an-eslint-config-plugin) for another PR!

File: lib/configs/recommended.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    'rest-spread-spacing': 'error',
+    'root/integration-drivers-return-this-or-null': 'error',
+    'root/integration-driver-import': 'error',
+    'root/integration-test-format': 'error',
+    'root/prefer-root-text': 'error',
+    'root/prevent-use-of-window-location': 'error',
+    'root/preceded-by-await': ['error', { 'functionNames': ['pollForCondition', 'wait', 'waitForElement'] }],
+    'root/sort-reducers-index': 'error',
+    'semi-spacing': 'error',
+    'semi': 'error',
```
Comment (JustAboutJeff) [inline lib/configs/recommended.js:87]: I would love to see this rule go away 😅

Comment (williams4027) [reply lib/configs/recommended.js:87]: Luckily, I am one of those people that loves semicolons and that was a previously enforced rule across projects. :)

**Review by kylekthompson — APPROVED** [review]
✨

**Pull request merged**